utilities/translation/clean.js

```diff
@@ -0,0 +1,185 @@
+
+// getting all .html file with their full path
+fileNames = getTemplates(initialRoute, [], false);
+
+// processing all .html file to isolate every key used in the template
+fileNames.forEach((item) => {
+    // fetching the content of the file and store it to fileContent
+    let fileContent = fs.readFileSync(path.resolve(item), 'utf8');
+
+    // Loading the template with cheerio
+    let $ = cheerio.load(fileContent);
+
+    // looking for every element containing the attribute translate
+    $('*[translate]').each((i, elt) => {
+        lastElement = elt.children.pop();
+
+        if (lastElement) {
+            // will return the key string of the form KEY1.KEY2.KEY3 found in the .html file
+            keyString = lastElement.data;
+
+            if (keyString) {
+
+                // removing the last key and reassign the object to the original one
+                en[keyString.split('.')[0]] = removeLastKey(keyString, en);
+            }
+        }
+
+    });
+});
+
+// Now the en object will contain only used key, the next step is trying to remove them safely and through this script
+let remainKeys = Object.keys(en).sort();
+
+remainKeys.forEach((key) => {
+    // transforming to lower case every key and removing unuseful spaces
+    key = key.toLowerCase().trim();
+
+    // Evoiding dealing with value parsed in angular format and other
+    if (key.indexOf('{') === -1 &&
+        key.indexOf('}') === -1 &&
+        key.indexOf('$') === -1 &&
+        key !== '') {
+        try {
+            console.log('Processing', key, '.json', '...');
+            var jsonRoute = path.resolve(__dirname, i18nRoute, key + '.json');
+
+            //loading the small json file used for developpement
+            let obj = require(jsonRoute);
+
+            // for every i18n json file, this function will return an array of keyString with the format : KEY1.KEY2.KEY3
+            var list = getAllKeyString(obj, '', []);
+
+            //processing every single key string
+            list.forEach((item) => {
+                // return true if the key string is present in the object containing data to remove (en.json)
+                if (isPresent(item)) {
+
+                    //if true, the last key is removed from the small json file obj
+                    obj[item.split('.')[0]] = removeLastKey(item, obj);
+                }
+            });
+
+            // stringify the object in order to write with utf8 encoding
+            const objAsString = JSON.stringify(obj);
+            const f = fs.openSync(jsonRoute, 'w');
+
+            // Writting  a new content into the small json file i18n
+            fs.writeFileSync(f, objAsString, 'utf8');
+        }
+        catch (e) {
+            //FIX ME : in case require can not find the file, how can I handle it properly?
+        }
+    }
+});
+
+function getAllKeyString(obj, path, res) {
+    res = res || [];
+
+    keys = Object.keys(obj).sort();
+
+    keys.forEach((key) => {
```

Comment:
It looks like you might be able to do:
```js
return keys.map(key => {
});
```
Then instead of `return res.concat(x)` it would just be `return x`.